examples/body_types.rs

```diff
@@ -0,0 +1,36 @@
+#![feature(async_await, futures_api)]
+
+#[macro_use]
+extern crate serde_derive;
+use tide::body;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+struct Message {
+  author: Option<String>,
+  contents: String,
+}
+
+async fn echo_string(msg: String) -> String {
+  println!("String: {}", msg);
```

Comment:
I think we shouldn't do `println!()` in examples, rather I think it would be nice to show an example with logging.

Reply:
Agreed! Logging middleware hasn't been implemented yet. Should I implement one in the example? Or should we wait until that is done to do that here?

Reply:
Yeah, I just saw that on https://github.com/rust-net-web/tide/issues/8. I would also like to try out experimenting with logging if I can. Maybe can just leave this be for now.